api/pkg/apis/projectcalico/v3/bgppeer.go

```diff
@@ -96,6 +96,9 @@ type BGPPeerSpec struct {
 	// Time to allow for software restart.  When specified, this is configured as the graceful
 	// restart timeout.  When not specified, the BIRD default of 120s is used.
 	MaxRestartTime *metav1.Duration `json:"maxRestartTime,omitempty"`
+	// Maximum number of local AS numbers that are allowed in the AS path for received routes.
+	// This removes BGP loop prevention and should only be used if absolutely necesssary.
+	NumAllowedLocalASNumbers *int32 `json:"numAllowedLocalASNumbers,omitempty"`
```

Comment:
nit, but should have `// +optional` against it (so should a bunch of these fields, tbh).

I think we run in "optional by default" mode anyway, but it's good to be explicit.